service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/C21OrderEmailContentProviderTest.java

```diff
@@ -38,71 +39,102 @@
 
+        given(cafcassLookupConfiguration.getCafcass(LOCAL_AUTHORITY_CODE))
+            .willReturn((new CafcassLookupConfiguration.Cafcass(CAFCASS_NAME, CAFCASS_EMAIL)));
+
         familyManCaseNumber = RandomStringUtils.randomAlphabetic(8);
+        documentId = randomUUID().toString();
+        subjectLine = "Jones, " + familyManCaseNumber;
     }
 
     @Test
     void shouldReturnExactC21NotificationParametersWithUploadedDocumentUrl() {
+        final String documentUrl = "http://dm-store:8080/documents/" + documentId + "/binary";
         CaseDetails caseDetails = createCaseDetailsWithSingleC21Element();
 
-        final String subjectLine = "Jones, " + familyManCaseNumber;
+        Map<String, Object> returnedCafcassParameters =
+            c21OrderEmailContentProvider.buildC21OrderNotificationParametersForCafcass(
+                caseDetails, LOCAL_AUTHORITY_CODE, documentUrl);
 
-        Map<String, Object> returnedParameters = c21OrderEmailContentProvider.buildC21OrderNotification(caseDetails,
-            LOCAL_AUTHORITY_CODE);
+        assertThat(returnedCafcassParameters)
+            .extracting("subjectLine", "localAuthorityOrCafcass", "hearingDetailsCallout",
+                "linkToDocument", "reference", "caseUrl")
+            .containsExactly(subjectLine, "Example Cafcass",
+                (subjectLine + ", hearing " + dateFormatterService.formatLocalDateToString(today, FormatStyle.MEDIUM)),
+                documentUrl, "167888", "/case/" + JURISDICTION + "/" + CASE_TYPE + "/167888");
 
-        assertThat(returnedParameters)
-            .extracting("subjectLine", "localAuthorityOrCafcass", "hearingDetailsCallout", "reference",
-                 "caseUrl")
-            .containsExactly(subjectLine, "Example Local Authority",
-                subjectLine + ", hearing " + dateFormatterService.formatLocalDateToString(date, FormatStyle.MEDIUM),
-                "167888", "/case/" + JURISDICTION + "/" + CASE_TYPE + "/167888");
+        Map<String, Object> returnedLocalAuthorityParameters =
+            c21OrderEmailContentProvider.buildC21OrderNotificationParametersForLocalAuthority(
+                caseDetails, LOCAL_AUTHORITY_CODE, documentUrl);
 
-        /*  extracting separately as document_binary_url in populateCaseDetailsWithSingleC21Element()
-            method uses UUID.randomUUID()
-         */
-        assertThat(returnedParameters).extracting("linkToDocStore").isNotEqualTo("");
+        assertThat(returnedLocalAuthorityParameters)
+            .extracting("subjectLine", "localAuthorityOrCafcass", "hearingDetailsCallout",
+                "linkToDocument", "reference", "caseUrl")
+            .containsExactly(subjectLine, "Example Local Authority",
+                (subjectLine + ", hearing " + dateFormatterService.formatLocalDateToString(today, FormatStyle.MEDIUM)),
+                documentUrl, "167888", "/case/" + JURISDICTION + "/" + CASE_TYPE + "/167888");
```

Comment:
Isn't this another test as we are checking something else? The unit test could assert one thing.